check_links/mod.ts

```diff
@@ -0,0 +1,46 @@
+import { merge } from "lume/core/utils/object.ts";
+import type Site from "lume/core/site.ts";
+import modifyUrls from "lume/plugins/modify_urls.ts";
+
+// todo:
+// - make file extension configurable
+// - make code simpler
+
+export interface Options {
+}
+
+export const defaults: Options = {};
+
+/**
+ * This plugin check broken links in *.html output files.
+ */
+export default function (userOptions?: Options) {
+  const options = merge(defaults, userOptions);
+
+  return (site: Site) => {
+    site.use(modifyUrls({
+      fn(url, page, _element) {
+        const url_base = site.options.location;
+        const urls: string[] = [];
+        for (const page of site.pages) {
```

Comment:
This loop is executed for every url found in the html code. It should be moved outside this function and run only once. For example:

```js
const urls = new Set<string>(); // Set is more performant than arrays

site.process("*", (pages) => {
  url.clear(); // Clear on rebuild
  for (const page of pages) {
    urls.add(site.url(page.data.url, true)); // site.url() return the full url if the second argument is true
  }
  for (const file of site.files) {
    urls.add(site.url(file.outputPath, true));
  }
});

site.use(modifyUrls({
  // Use the urls
})
```